src/core/trtserver2.h

```diff
@@ -304,6 +305,42 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestOutputData(
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestRemoveAllOutputs(
     TRTSERVER2_InferenceRequest* inference_request);
 
+///
+/// TRTSERVER2_ModelIndex
+///
+/// Object representing model indices.
+///
+
+/// Get the object representing the indices of all unique models in the
+/// model repository.
+/// \param server The inference server object.
+/// \param model_indices Returns the TRTSERVER2_ModelIndex object that
+/// is used to manage the lifecycle of the returned strings.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_ServerModelIndex(
+    TRTSERVER_Server* server, TRTSERVER2_ModelIndex** model_indices);
+
+/// Get the names of all unique models in the model repository.
+/// The caller does not own the returned strings and must not modify or
+/// delete them. The lifetime of the returned strings extends only as
+/// long as 'model_indices' and must not be accessed once 'model_indices'
+/// is deleted.
+/// \param model_indices The TRTSERVER2_ModelIndex object that
```

Comment:
All the places you call this model_indices is confusing, because TRTSERVER2_ModelIndex is a single index. It is an index that contains all the models in the repository(s). Currently it only has the names of the models but in the future it might have more index information. So "\param model_index The index of models".